x-pack/platform/plugins/shared/streams_app/public/components/data_management/stream_detail_enrichment/processor_status_indicator.tsx

```diff
@@ -0,0 +1,162 @@
+ * 2.0.
+ */
+
+import {
+  EuiAvatar,
+  EuiAvatarProps,
+  EuiLoadingSpinner,
+  EuiToolTip,
+  useEuiTheme,
+} from '@elastic/eui';
+import { useSelector } from '@xstate5/react';
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import { ProcessorConfigurationProps } from './processors';
+import {
+  useSimulatorSelector,
+  useStreamEnrichmentSelector,
+} from './state_management/stream_enrichment_state_machine';
+import { selectWhetherAnyProcessorBeforePersisted } from './state_management/stream_enrichment_state_machine/selectors';
+
+export const ProcessorStatusIndicator = ({
+  processorRef,
+}: {
+  processorRef: ProcessorConfigurationProps['processorRef'];
+}) => {
+  const { euiTheme } = useEuiTheme();
+
+  const isNew = useSelector(processorRef, (snapshot) => snapshot.context.isNew);
+
+  const hasSimulation = useSimulatorSelector((snapshot) => Boolean(snapshot.context.simulation));
+  const isParticipatingInSimulation = useSimulatorSelector((snapshot) =>
+    snapshot.context.processors.find((p) => p.id === processorRef.id)
+  );
+  const isSimulationRunning = useSimulatorSelector((snapshot) =>
+    snapshot.matches('runningSimulation')
+  );
+  const isPending = useSimulatorSelector(
+    (snapshot) =>
+      snapshot.context.simulation &&
+      !snapshot.context.simulation.processors_metrics[processorRef.id]
+  );
+  const isFailing = useSimulatorSelector((snapshot) =>
+    Boolean(
+      snapshot.context.simulation?.processors_metrics[processorRef.id]?.errors.some(
+        (e) => e.type === 'generic_simulation_failure'
+      )
+    )
+  );
+  const isAnyProcessorBeforePersisted = useStreamEnrichmentSelector(
+    selectWhetherAnyProcessorBeforePersisted
+  );
+
+  let variant:
+    | (EuiAvatarProps & {
+        tooltipContent?: string;
+      })
+    | null = null;
+
+  if (isAnyProcessorBeforePersisted) {
+    const name = i18n.translate(
+      'xpack.streams.streamDetailView.managementTab.enrichment.mixedProcessorTooltip',
+      {
+        defaultMessage:
+          'Currently, is not possible to simulate persisted and new processors together. Move the new processors after the persisted processors to restore the simulation behavior.',
+      }
+    );
+    variant = {
+      name,
+      iconType: 'dot',
+      color: euiTheme.colors.backgroundBaseDisabled,
+      tooltipContent: name,
+    };
+  } else if (!isParticipatingInSimulation) {
+    const name = isNew
+      ? i18n.translate(
+          'xpack.streams.streamDetailView.managementTab.enrichment.skippedNewProcessorTooltip',
+          {
+            defaultMessage:
+              'This processor is skipped since it comes after the processor under edit.',
+          }
+        )
+      : i18n.translate(
+          'xpack.streams.streamDetailView.managementTab.enrichment.skippedConfiguredProcessorTooltip',
+          {
+            defaultMessage: 'This processor is skipped since it is persisted already.',
```

Comment:
```suggestion
            defaultMessage: 'Processor skipped because it already exists.',
```
Not sure exactly what this one means, but I gave it a shot. Does this mean it's a duplicate of an existing processor?

Reply:
```suggestion
            defaultMessage: 'Processor skipped because it was created in a previous simulation session.',
```
how about this.